File: Ruby on Rails/app/controllers/healthcheck_controller.rb
require 'rest-client'

class HealthcheckController < ApplicationController
  MOCK_SERVER_URL = 'https://interview.riskxint.com'.freeze

  def index
    mock_server_response = RestClient.get("#{MOCK_SERVER_URL}/healthcheck", nil)
    puts("Mock server responded to health check with the following response: #{mock_server_response.body}")

    render json: { status: 'OK' }, status: 200
  end
end

File: C#/Controllers/HealthcheckController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System.Net.Http;

namespace paymentGatewaySimulation.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class HealthcheckController : ControllerBase
    {
        public const string MockServiceUrl = "https://interview.riskxint.com";

        private readonly ILogger<HealthcheckController> _logger;

        public HealthcheckController(ILogger<HealthcheckController> logger)
        {
            _logger = logger;
        }

        [HttpGet]
        public IActionResult Get()
        {
            using var client = new HttpClient();
            var response = client.GetAsync(MockServiceUrl + "/healthcheck").Result;
            return Ok("Mock server responded to health check with response: " + response.ReasonPhrase);
        }
    }
}
